numpy/core/src/multiarray/descriptor.c

```diff
@@ -1661,7 +1655,11 @@ _convert_from_bytes(PyObject *obj)
                 }
             }
         }
-        return _convert_from_any(item);
+        /*
+         * Probably only ever dispatches to `_convert_from_type`, but who
+         * knows what users are injecting into `np.typeDict`.
+         */
+        return _convert_from_any(item, align);
```

Comment:
If we want to keep the old behavior, we could change this to
```suggestion
        return _convert_from_any(item, 0);
```